From 8c17e8cb46b9cfb92b12b7ccef88c7792123eb96 Mon Sep 17 00:00:00 2001
From: Remi Collet <remi@remirepo.net>
Date: Wed, 12 Feb 2020 08:09:10 +0100
Subject: [PATCH] install kms_caller_identity_request.h

---
 kms-message/CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/kms-message/CMakeLists.txt b/kms-message/CMakeLists.txt
index fae39bd..45c2060 100644
--- a/kms-message/CMakeLists.txt
+++ b/kms-message/CMakeLists.txt
@@ -139,6 +139,7 @@ install (
 install (
    FILES
    src/kms_message/kms_b64.h
+   src/kms_message/kms_caller_identity_request.h
    src/kms_message/kms_decrypt_request.h
    src/kms_message/kms_encrypt_request.h
    src/kms_message/kms_message.h
-- 
2.24.1

