From 6a55c3123afa4196eb432ae81fddeced7131524f Mon Sep 17 00:00:00 2001
From: Remi Collet <remi@remirepo.net>
Date: Fri, 5 Nov 2021 09:06:34 +0100
Subject: [PATCH] MONGOCRYPT-359 simplify HMAC creation, fix compatibility with
 OpenSSL 3.0

---
 src/crypto/libcrypto.c | 17 ++++++++++++++---
 1 file changed, 14 insertions(+), 3 deletions(-)

diff --git a/src/crypto/libcrypto.c b/src/crypto/libcrypto.c
index 61d1c0f6..15613620 100644
--- a/src/crypto/libcrypto.c
+++ b/src/crypto/libcrypto.c
@@ -190,14 +190,24 @@ _native_crypto_hmac_sha_512 (const _mongocrypt_buffer_t *key,
                              mongocrypt_status_t *status)
 {
    const EVP_MD *algo;
-   HMAC_CTX *ctx;
-   bool ret = false;
 
-   ctx = HMAC_CTX_new ();
    algo = EVP_sha512 ();
    BSON_ASSERT (EVP_MD_block_size (algo) == 128);
    BSON_ASSERT (EVP_MD_size (algo) == MONGOCRYPT_HMAC_SHA512_LEN);
 
+#if OPENSSL_VERSION_NUMBER >= 0x10100000L
+   if (!HMAC(algo, key->data, key->len, in->data, in->len, out->data, NULL /* unused out len */)) {
+      CLIENT_ERR ("error initializing HMAC: %s",
+                  ERR_error_string (ERR_get_error (), NULL));
+      return false;
+   }
+   return true;
+#else
+   HMAC_CTX *ctx;
+   bool ret = false;
+
+   ctx = HMAC_CTX_new ();
+
    if (out->len != MONGOCRYPT_HMAC_SHA512_LEN) {
       CLIENT_ERR ("out does not contain %d bytes", MONGOCRYPT_HMAC_SHA512_LEN);
       return false;
@@ -225,6 +235,7 @@ _native_crypto_hmac_sha_512 (const _mongocrypt_buffer_t *key,
 done:
    HMAC_CTX_free (ctx);
    return ret;
+#endif
 }
 
 
