From e520f2b12c2750bca142dd118f58c932520e1891 Mon Sep 17 00:00:00 2001
From: Remi Collet <remi@remirepo.net>
Date: Mon, 6 Feb 2023 16:49:51 +0100
Subject: [PATCH] MONGOCRYPT-532 Fix i686 (32-bit) build

---
 cmake/IntelDFP.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmake/IntelDFP.cmake b/cmake/IntelDFP.cmake
index fb7714862..fe6e749c7 100644
--- a/cmake/IntelDFP.cmake
+++ b/cmake/IntelDFP.cmake
@@ -332,7 +332,7 @@ set_property (TARGET intel_dfp_obj PROPERTY POSITION_INDEPENDENT_CODE TRUE)
 
 # DFP needs information about the build target platform. Compute that:
 set (proc_lower $<LOWER_CASE:${CMAKE_SYSTEM_PROCESSOR}>)
-set (ia32_list i386 emscripted x86 arm)
+set (ia32_list i386 i486 i586 i686 pentium3 pentium4 athlon geode emscripted x86 arm)
 set (efi2_list aarch64 arm64 x86_64 ppc64le riscv64)
 
 set (is_linux $<PLATFORM_ID:Linux>)
